File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/RevocationApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.VerifiableCredential
import my.ssdid.sdk.domain.revocation.RevocationManager
import my.ssdid.sdk.domain.revocation.RevocationStatus

class RevocationApi internal constructor(private val manager: RevocationManager) {
    suspend fun checkStatus(credential: VerifiableCredential): RevocationStatus =
        manager.checkRevocation(credential)
    fun invalidateCache() = manager.invalidateCache()
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/RevocationApi.swift
import Foundation

/// Facade for credential revocation checking.
public struct RevocationApi {
    let manager: RevocationManager

    public func checkStatus(_ credential: VerifiableCredential) async -> RevocationStatus {
        await manager.checkRevocation(credential)
    }

    public func invalidateCache() async {
        await manager.invalidateCache()
    }
}
